### PR TITLE
chore: fix tabs-demo style in compact mode

### LINKED ISSUE #24920: tabs-demo-card-top 样式紧凑模式下异常
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://ant.design/components/tabs-cn/#components-tabs-demo-card-top](https://ant.design/components/tabs-cn/#components-tabs-demo-card-top)

### 正常模式下

image

### 紧凑模式下

image

---

应该是需要修改 Demo 中自定义 css

### PR DESCRIPTION
**Pull request by AshoneA**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
Close #24920

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/tabs/demo/card-top.md](https://github.com/AshoneA/ant-design/blob/chore/tabs-demo/components/tabs/demo/card-top.md)

### CODE REVIEW

File: components/tabs/demo/card-top.md [hunk 1]
```diff
@@ -57,11 +57,13 @@ ReactDOM.render(
...
 }
-.card-container > .ant-tabs-card .ant-tabs-tab {
+.card-container > .ant-tabs-card .ant-tabs-tab,
+[data-theme='compact'] .card-container > .ant-tabs-card .ant-tabs-tab {
   border-color: transparent;
   background: transparent;
 }
-.card-container > .ant-tabs-card .ant-tabs-tab-active {
+.card-container > .ant-tabs-card .ant-tabs-tab-active,
+[data-theme='compact'] .card-container > .ant-tabs-card .ant-tabs-tab-active {
```
Comment (xrkffgg) [inline components/tabs/demo/card-top.md:66]: 我看这块下面 用<style> 包了一些样式，要不要把 2 个合一起，都放到上面来。

Comment (zombieJ) [reply components/tabs/demo/card-top.md:66]: @AshoneA 之前改这个的时候我也有点拿不准，为什么要这么写。确认一下合了会不会对切换有影响。

Comment (AshoneA) [reply components/tabs/demo/card-top.md:66]: 应该没啥区别，我看编译之后的样式都嵌套在 `<style>...</style>` 中。

File: components/tabs/demo/card-top.md [hunk 2]
```diff
@@ -57,38 +57,37 @@ ReactDOM.render(
 .card-container > .ant-tabs-card > .ant-tabs-nav::before {
   display: none;
 }
-.card-container > .ant-tabs-card .ant-tabs-tab {
+.card-container > .ant-tabs-card .ant-tabs-tab,
+[data-theme='compact'] .card-container > .ant-tabs-card .ant-tabs-tab {
   border-color: transparent;
```
Comment (xrkffgg) [inline components/tabs/demo/card-top.md:62]: 这块 是 dark 是不是都一样的

Comment (AshoneA) [reply components/tabs/demo/card-top.md:62]: 看以前写的好像是不一样， dark 模式下具体表现我就不是很清楚了。

File: components/tabs/demo/card-top.md (hunk 1 above)
Comment (zombieJ) [reply components/tabs/demo/card-top.md:66]: 👌 测一下如果没问题直接就可以合了

**Review by zombieJ — APPROVED** [review]

**Pull request merged**